src/main/java/server/Router.java

```diff
@@ -0,0 +1,37 @@
+package main.java.server;
+
+import main.java.application.Handler;
+
+import java.util.HashMap;
+import java.util.TreeMap;
+
+import static main.java.server.HTTPMessageComponents.HTTPMethods.*;
+
+public class Router {
+    public TreeMap<String, HashMap<String, Handler>> routes;
+
+    public Router() { this.routes = new TreeMap<>(); }
+
+    public TreeMap<String, HashMap<String, Handler>> routes() { return routes; }
+
+    public void get(String route, Handler handler) { addToRoutes(GET, route, handler); }
+
+    public void head(String route, Handler handler) { addToRoutes(HEAD, route, handler); }
+
+    public void options(String route, Handler handler) {
+        addToRoutes(OPTIONS, route, handler);
+    }
+
+    public void put(String route, Handler handler) {
+        addToRoutes(PUT, route, handler);
+    }
+
+    public void post(String route, Handler handler) {
+        addToRoutes(POST, route, handler);
+    }
+
+    private void addToRoutes(String method, String route, Handler handler) {
```

Comment:
How are we defining `route` in this namespace? If there are multiple definitions, could we pick some different names that reveal subtle distinctions between concepts?

Reply:
True - going forward I'm going to use `route` as a the larger journey a request takes (as in a `path` + `method(s)` + `handler`) and then rename everything else as a `path`.